backend/routes/songs.js

```diff
@@ -23,11 +23,23 @@ app.post("", (req, res, next) => {
 });
 
 app.get("", (req, res, next) => {
-    Song.find().then(songsResult => {
+    const pageSize = +req.query.pageSize;
+    const currPage = +req.query.page;
+    let fetchedSongs;
+    const songQuery = Song.find();
```

Comment:
What is Song. Find?

Reply:
And why it returns a promise?

Reply:
what is promise?

Reply:
the find is for change the query. It adds more functions before actually run the query.

Reply:
Where do we add these funcs?